geopandas/tests/test_sindex.py

```diff
@@ -172,10 +174,284 @@ def test_intersection_invalid_bounds_tuple(self, test_geom):
+                    "https://github.com/geopandas/geopandas/issues/1337"
+                )
+            raise e
+
+    # ------------------------- `query_bulk` tests -------------------------- #
+    @pytest.mark.parametrize(
+        "predicate, test_geom, expected",
+        (
+            (None, (-1, -1, -0.5, -0.5), [[], []]),
+            (None, (-0.5, -0.5, 0.5, 0.5), [[0], [0]]),
+            (None, (0, 0, 1, 1), [[0, 0], [0, 1]]),
+            ("intersects", (-1, -1, -0.5, -0.5), [[], []]),
+            ("intersects", (-0.5, -0.5, 0.5, 0.5), [[0], [0]]),
+            ("intersects", (0, 0, 1, 1), [[0, 0], [0, 1]]),
+            ("within", (0.25, 0.28, 0.75, 0.75), [[], []]),  # does not intersect
+            ("within", (0, 0, 10, 10), [[], []]),  # intersects but is not within
+            ("within", (11, 11, 12, 12), [[0], [5]]),  # intersects and is within
+            ("contains", (0, 0, 1, 1), [[], []]),  # intersects but does not contain
+            (
+                "contains",
+                (0, 0, 1.001, 1.001),
+                [[0], [1]],
+            ),  # intersects 2 and contains 1
+            (
+                "contains",
+                (0.5, 0.5, 1.001, 1.001),
+                [[0], [1]],
+            ),  # intersects 1 and contains 1
+            ("contains", (0.5, 0.5, 1.5, 1.5), [[0], [1]]),  # intersects and contains
+            (
+                "contains",
+                (-1, -1, 2, 2),
+                [[0, 0], [0, 1]],
+            ),  # intersects and contains multiple
+        ),
+    )
+    def test_query_bulk(self, predicate, test_geom, expected):
+        """Tests the `query_bulk` method with valid
+        inputs and valid predicates.
+        """
+        # pass through GeoSeries to have GeoPandas
+        # determine if it should use shapely or pygeos geometry objects
+        test_geom = geopandas.GeoSeries([box(*test_geom)], index=["0"])
+
+        res = self.df.sindex.query_bulk(test_geom, predicate=predicate)
+        assert_array_equal(res, expected)
+
+    @pytest.mark.parametrize(
+        "test_geoms, expected_value",
+        [
+            # single empty geometry
+            ([GeometryCollection()], [[], []]),
+            # None should be skipped
+            ([GeometryCollection(), None], [[], []]),
+            ([None], [[], []]),
+            ([None, box(-0.5, -0.5, 0.5, 0.5), None], [[1], [0]]),
+        ],
+    )
+    def test_query_bulk_empty_geometry(self, test_geoms, expected_value):
+        """Tests the `query_bulk` method with an empty geometry.
+        """
+        # pass through GeoSeries to have GeoPandas
+        # determine if it should use shapely or pygeos geometry objects
+        # note: for this test, test_geoms (note plural) is a list already
+        test_geoms = geopandas.GeoSeries(test_geoms, index=range(len(test_geoms)))
+        res = self.df.sindex.query_bulk(test_geoms)
+        assert_array_equal(res, expected_value)
+
+    @pytest.mark.parametrize(
```

Comment:
No need for `parametrize` here, additional variants are unlikely.

Reply:
I tend to like keeping the `parametrize` because it makes the test cases more explicit instead of burying them into the test function body. Is there is another way to do this for a single testcase?

Reply:
It's usually just a matter of personal preference, at that level.  Personally I fully expect that the details of the test are contained in the test function, unless there is a need to iterate over several input parameters.  Having it `parametrize` is a hint that the implementation should be tested over many variants, not having `parametrize` is a hint that a single variant is sufficient.

Reply:
Ok, @jorisvandenbossche seemed to be of the same opinion as you; I went ahead and removed the parametrizations for single inputs.